File: android/app/src/debug/java/com/laneshadow/sandbox/mockproviders/SessionsMockProvider.kt
package com.laneshadow.sandbox.mockproviders

/**
 * Mock provider for Sessions screen data.
 *
 * Provides list of past and active ride sessions
 * for the Sessions/history drawer screen.
 */
object SessionsMockProvider : MockProvider<SessionsScreenState> {

    override val variants: List<String> = listOf(
        "default", "empty", "overflow", "long-copy",
        "s05-new-confirm", "s04-grouped"
    )

    override fun value(variant: String): SessionsScreenState {
        return when (variant) {
            "empty" -> emptyState()
            "overflow" -> overflowState()
            "long-copy" -> longCopyState()
            "s05-new-confirm" -> s05NewConfirmState()
            "s04-grouped" -> s04GroupedState()
            else -> defaultState()
        }
    }

    private fun defaultState(): SessionsScreenState {
        val sessions = listOf(
            Session(
                id = "session-001",
                title = "Santa Cruz loop",
                preview = "Scenic 2-hour ride, avoid highways",
                meta = "3 routes · Active",
                `when` = "Now",
                active = true,
                routeIds = listOf("route-001", "route-002", "route-003"),
                createdAt = "2026-04-25T10:30:00Z"
            ),
            Session(
                id = "session-002",
                title = "Niles Canyon morning",
                preview = "Twisty roads, coffee stop in Sunol",
                meta = "2 routes",
                `when` = "Tue",
                active = false,
                routeIds = listOf("route-004", "route-005"),
                createdAt = "2026-04-22T08:15:00Z"
            ),
            Session(
                id = "session-003",
                title = "Diablo summit attempt",
                preview = "Early morning climb, light traffic",
                meta = "1 route",
                `when` = "Apr 12",
                active = false,
                routeIds = listOf("route-008"),
                createdAt = "2026-04-12T06:45:00Z"
            )
        )

        return SessionsScreenState(
            sessions = sessions,
            activeSessionId = "session-001"
        )
    }

    private fun emptyState(): SessionsScreenState {
        return SessionsScreenState(
            sessions = emptyList(),
            activeSessionId = null
        )
    }

    private fun overflowState(): SessionsScreenState {
        val sessions = (1..12).map { i ->
            Session(
                id = "session-00$i",
                title = "Session $i",
                preview = "Preview text for session $i",
                meta = "${i} routes",
                `when` = when (i) {
                    1 -> "Now"
                    2 -> "Tue"
                    3 -> "Apr 12"
                    else -> "Apr ${12 - i}"
                },
                active = i == 1,
                routeIds = listOf("route-$i"),
                createdAt = "2026-04-${25 - i}T10:30:00Z"
            )
        }

        return SessionsScreenState(
            sessions = sessions,
            activeSessionId = "session-001"
        )
    }

    private fun longCopyState(): SessionsScreenState {
        val sessions = listOf(
            Session(
                id = "session-001",
                title = "The Ultimate Santa Cruz Coastal Loop Adventure with Multiple Scenic Stops and Optional Extensions",
                preview = "I am looking for a truly scenic and memorable two-hour motorcycle ride that takes me down the coast to Santa Cruz while completely avoiding all highways and major roads. I want to see the ocean, ride through redwood forests, and maybe stop for coffee along the way. Please include optional detours to points of interest.",
                meta = "3 routes · Active · Plus 2 optional extensions",
                `when` = "Now",
                active = true,
                routeIds = listOf("route-001", "route-002", "route-003", "route-012", "route-007"),
                createdAt = "2026-04-25T10:30:00Z"
            ),
            Session(
                id = "session-002",
                title = "Niles Canyon Morning Coffee Run Through The East Bay Hills",
                preview = "An early morning adventure through the twisty roads of Niles Canyon with a planned coffee stop at the famous Sunol Coffee Shop where local riders gather every weekend to share stories and plan their next rides",
                meta = "2 routes · Active",
                `when` = "Tue",
                active = false,
                routeIds = listOf("route-004", "route-005"),
                createdAt = "2026-04-22T08:15:00Z"
            ),
            Session(
                id = "session-003",
                title = "Mount Hamilton Summit Challenge",
                preview = "Early morning climb",
                meta = "1 route",
                `when` = "Apr 12",
                active = false,
                routeIds = listOf("route-008"),
                createdAt = "2026-04-12T06:45:00Z"
            )
        )

        return SessionsScreenState(
            sessions = sessions,
            activeSessionId = "session-001"
        )
    }

    /**
     * S05 new-confirm state — shows "Start a new ride?" confirm dialog
     */
    private fun s05NewConfirmState(): SessionsScreenState {
        val sessions = listOf(
            Session(
                id = "session-001",
                title = "Santa Cruz loop",
                preview = "Scenic 2-hour ride, avoid highways",
                meta = "3 routes · Active",
                `when` = "Now",
                active = true,
                routeIds = listOf("route-001", "route-002", "route-003"),
                createdAt = "2026-04-25T10:30:00Z"
            )
        )

        return SessionsScreenState(
            sessions = sessions,
            activeSessionId = "session-001",
            showConfirmDialog = true
        )
    }

    /**
     * S04 grouped state — shows sessions grouped by date
     */
    private fun s04GroupedState(): SessionsScreenState {
        val tonight = listOf(
            Session(
                id = "session-001",
                title = "Sunset coastal run",
                preview = "Evening ride along the coast",
                meta = "2 routes · Active",
                `when` = "TONIGHT",
                active = true,
                routeIds = listOf("route-001", "route-002"),
                createdAt = "2026-04-28T18:00:00Z"
            )
        )

        val today = listOf(
            Session(
                id = "session-002",
                title = "Morning coffee run",
                preview = "Niles Canyon with coffee stop",
                meta = "1 route",
                `when` = "TODAY",
                active = false,
                routeIds = listOf("route-003"),
                createdAt = "2026-04-28T08:00:00Z"
            ),
            Session(
                id = "session-003",
                title = "Lunch loop",
                preview = "Quick spin through the hills",
                meta = "1 route",
                `when` = "TODAY",
                active = false,
                routeIds = listOf("route-004"),
                createdAt = "2026-04-28T12:00:00Z"
            )
        )

        val thisWeek = listOf(
            Session(
                id = "session-004",
                title = "Diablo summit",
                preview = "Early morning climb",
                meta = "1 route",
                `when` = "THIS WEEK",
                active = false,
                routeIds = listOf("route-005"),
                createdAt = "2026-04-25T06:00:00Z"
            ),
            Session(
                id = "session-005",
                title = "Santa Cruz scenic",
                preview = "Coastal highway cruise",
                meta = "3 routes",
                `when` = "THIS WEEK",
                active = false,
                routeIds = listOf("route-006", "route-007", "route-008"),
                createdAt = "2026-04-24T10:00:00Z"
            )
        )

        val lastWeek = listOf(
            Session(
                id = "session-006",
                title = "Wine country tour",
                preview = "Backroads through Napa",
                meta = "2 routes",
                `when` = "LAST WEEK",
                active = false,
                routeIds = listOf("route-009", "route-010"),
                createdAt = "2026-04-15T09:00:00Z"
            )
        )

        val sections = listOf(
            com.laneshadow.sandbox.mockproviders.SessionSection(label = "TONIGHT", sessions = tonight),
            com.laneshadow.sandbox.mockproviders.SessionSection(label = "TODAY", sessions = today),
            com.laneshadow.sandbox.mockproviders.SessionSection(label = "THIS WEEK", sessions = thisWeek),
            com.laneshadow.sandbox.mockproviders.SessionSection(label = "LAST WEEK", sessions = lastWeek)
        )

        val allSessions = tonight + today + thisWeek + lastWeek

        return SessionsScreenState(
            sessions = allSessions,
            activeSessionId = "session-001",
            sections = sections
        )
    }
}

File: ios/LaneShadow/Sandbox/MockProviders/SessionsMockProvider.swift
import Foundation

/// Mock provider for Sessions screen data.
///
/// Provides a list of past and active sessions
/// for the Sessions/history screen.
public enum SessionsMockProvider: MockProvider {
    public static let variants = ["default", "empty", "overflow", "long-copy"]

    public static func value(variant: String = "default") -> SessionsScreenState {
        switch variant {
        case "empty":
            emptyState()
        case "overflow":
            overflowState()
        case "long-copy":
            longCopyState()
        default:
            defaultState()
        }
    }

    private static func defaultState() -> SessionsScreenState {
        SessionsScreenState(
            sessions: [
                Session(
                    id: "session-003",
                    title: "Santa Cruz loop",
                    preview: "Twisty roads down to Santa Cruz and back via 9",
                    meta: "3 routes · Active",
                    when: "Now",
                    active: true,
                    routeIds: ["route-001", "route-002", "route-003"],
                    createdAt: "2025-04-25T10:30:00Z"
                ),
                Session(
                    id: "session-002",
                    title: "Big Sur weekend",
                    preview: "Coastal cruise with overnight stay",
                    meta: "2 routes",
                    when: "Tue",
                    active: false,
                    routeIds: ["route-004", "route-005"],
                    createdAt: "2025-04-22T14:20:00Z"
                ),
                Session(
                    id: "session-001",
                    title: "Wine country tour",
                    preview: "Napa Valley backroads",
                    meta: "4 routes",
                    when: "Apr 12",
                    active: false,
                    routeIds: ["route-006", "route-007", "route-008", "route-009"],
                    createdAt: "2025-04-12T09:15:00Z"
                ),
            ],
            activeSessionId: "session-003"
        )
    }

    private static func emptyState() -> SessionsScreenState {
        SessionsScreenState(
            sessions: [],
            activeSessionId: nil
        )
    }

    private static func overflowState() -> SessionsScreenState {
        var sessions: [Session] = []

        let titles = [
            "Morning mountain run", "Coastal sunset cruise", "Wine country tour",
            "Redwood forest adventure", "Desert loop", "City escape",
            "Border run", "Pass storm chasing", "Peak bagging",
            "Valley cruise", "River road romp", "Lake loop tour",
        ]

        for (index, title) in titles.enumerated() {
            sessions.append(Session(
                id: "session-\(String(format: "%03d", index + 1))",
                title: title,
                preview: "Preview text for \(title.lowercased()) with some detail about the route",
                meta: "\(index + 1) route\(index == 0 ? "" : "s")",
                when: index < 3 ? "Now" : (index < 6 ? "Tue" : "Apr \(12 - index % 10)"),
                active: index == 0,
                routeIds: Array((1 ... index + 1).map { "route-\(String(format: "%03d", $0))" }),
                createdAt: "2025-04-25T10:30:00Z"
            ))
        }

        return SessionsScreenState(
            sessions: sessions,
            activeSessionId: "session-001"
        )
    }

    private static func longCopyState() -> SessionsScreenState {
        SessionsScreenState(
            sessions: [
                Session(
                    id: "session-003",
                    title: "Santa Cruz Coastal Loop with Technical Mountain Passes and Scenic Overlooks",
                    preview: "An epic adventure featuring the finest twisty roads leading down to Santa Cruz, including the legendary Stage Road with its perfectly engineered decreasing-radius corners, followed by a return journey via Highway 9 through redwood forests",
                    meta: "3 meticulously planned routes · Currently Active",
                    when: "Now",
                    active: true,
                    routeIds: ["route-001", "route-002", "route-003"],
                    createdAt: "2025-04-25T10:30:00Z"
                ),
                Session(
                    id: "session-002",
                    title: "Big Sur Weekend Adventure with Luxury Accommodations and Fine Dining",
                    preview: "A comprehensive two-day coastal cruiser featuring overnight accommodation at the historic Ventana Big Sur resort, with carefully planned stops at scenic overlooks, art galleries, and the world-famous Nepenthe restaurant perched cliffside above the Pacific Ocean",
                    meta: "2 distinct route options for each day",
                    when: "Tue",
                    active: false,
                    routeIds: ["route-004", "route-005"],
                    createdAt: "2025-04-22T14:20:00Z"
                ),
                Session(
                    id: "session-001",
                    title: "Napa Valley Wine Country Tour featuring Premium Tastings and Vineyard Lunches",
                    preview: "An exclusive curated journey through Napa Valley's most prestigious wineries, including private tastings at Opus One, Dominus, and Screaming Eagle, with a specially planned route through the Silverado Trail and lunch at the French Laundry",
                    meta: "4 unique routes covering different AVAs",
                    when: "Apr 12",
                    active: false,
                    routeIds: ["route-006", "route-007", "route-008", "route-009"],
                    createdAt: "2025-04-12T09:15:00Z"
                ),
            ],
            activeSessionId: "session-003"
        )
    }
}
